### PR TITLE
Remove some boilerplate

### PR DESCRIPTION
**Pull request by datalek**
#### Motivation and Context
This PR eliminates some of the boilerplate code.

#### How Has This Been Tested?

Unit tests

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/domain/SystemEnv.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  senderPAId: string;
+  // generators
+  iunGenerator: IO<Iun>;
+  dateGenerator: IO<Date>;
+  // repositories
+  createNotificationRequestRecordRepository: NewNotificationRepository;
+  findNotificationRequestRecordRepository: CheckNotificationStatusRecordRepository;
+  consumeEventStreamRecordRepository: ConsumeEventStreamRecordRepository;
+  getNotificationDetailRecordRepository: GetNotificationDetailRecordRepository;
+  getNotificationDocumentMetadataRecordRepository: GetNotificationDocumentMetadataRecordRepository;
```
Comment (kin0992) [inline src/domain/SystemEnv.ts:21]: Just a couple of questions about this:
why the repositories should be a part of the system environment?
And why the system env does not contain all the repositories? (e.g. the PreLoadRepository, ecc.)

Comment (datalek) [reply src/domain/SystemEnv.ts:21]: > Just a couple of questions about this: why the repositories should be a part of the system environment? 

The `SystemEnv` is born to keep some common dependencies on it, but I think neither the name nor the place are the best ones. Here are two options that should improve comprehension: 
1. Move `SystemEnv` from `domain` folder to `useCases` folder.
   - [Optional] Rename  `SystemEnv` to `UseCasesEnv`.
2. Split the `SystemEnv` into two `Env`, one for the domain (the `DomainEnv`) and one for the useCases (the `UseCasesEnv`), the latter will contain repositories and the `DomainEnv`.

> And why the system env does not contain all the repositories? (e.g. the PreLoadRepository, ecc.)

The goal of this refactor is to remove some code duplication, the repositories contained in this structure are those used in the use cases involved in this refactor.

Comment (kin0992) [reply src/domain/SystemEnv.ts:21]: Yeah, this work simplifies a lot the code, but speaking of _responsibility_, I think it's better to split the `SystemEnv` into two separate envs with the dependency you said.

Comment (datalek) [reply src/domain/SystemEnv.ts]: Done 🍰

**Review by kin0992 — APPROVED** [review]
🥟

**Review by kin0992 — APPROVED** [review]
💯

**Pull request merged**